stencila/pyla/servers.py

```diff
@@ -0,0 +1,289 @@
+"""
+Module for server classes.
+"""
+
+import sys
+
+import enum
+import json
+import logging
+import typing
+from socket import socket
+from stencila.schema.types import CodeChunk
+from stencila.schema.util import to_json, from_dict
+
+from .code_parsing import simple_code_chunk_parse
+from .interpreter import Interpreter
+
+StreamType = typing.Union[typing.BinaryIO, socket]
+
+
+def data_to_bytes(data: typing.Any) -> bytes:
+    """Convert `data` to `bytes`."""
+    return bytes((data,))
+
+
+def encode_int(number: int) -> bytes:
+    """Pack `number` into varint bytes"""
+    buf = b''
+    while True:
+        to_write = number & 0x7f
+        number >>= 7
+        if number:
+            buf += data_to_bytes(to_write | 0x80)
+        else:
+            buf += data_to_bytes(to_write)
+            break
+    return buf
+
+def read_one(stream: StreamType) -> int:
+    """Read a byte from the file (as an integer).
+
+    Raises EOFError if the stream ends while reading bytes.
+    """
+    char = stream_read(stream, 1)
+    if char == b'':
```

Comment:
This could be safer if it were `if not char`

Reply:
OK (this is unchanged from `master`: https://github.com/stencila/pyla/blob/master/stencila/pyla/listener.py#L91)